designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteFactory.scala

```diff
@@ -0,0 +1,464 @@
+import pl.touk.nussknacker.ui.process.processingtype.provider.ProcessingTypeDataProvider
+import pl.touk.nussknacker.ui.process.repository._
+import pl.touk.nussknacker.ui.process.repository.activities.ScenarioActivityRepository
+import pl.touk.nussknacker.ui.process.repository.stickynotes.DbStickyNotesRepository
+import pl.touk.nussknacker.ui.process.scenarioactivity.FetchScenarioActivityService
+import pl.touk.nussknacker.ui.process.version.{ScenarioGraphVersionRepository, ScenarioGraphVersionService}
+import pl.touk.nussknacker.ui.processreport.ProcessCounter
+import pl.touk.nussknacker.ui.security.api.{AuthenticationResources, AuthManager, LoggedUser, NussknackerInternalUser}
+import pl.touk.nussknacker.ui.statistics.{
+  FingerprintService,
+  PublicEncryptionKey,
+  StatisticUrlConfig,
+  UsageStatisticsReportsSettingsService
+}
+import pl.touk.nussknacker.ui.util._
+import pl.touk.nussknacker.ui.validation.ScenarioLabelsValidator
+
+import java.time.Clock
+import scala.concurrent.Future
+
+class AkkaHttpBasedRouteFactory(
+    clock: Clock,
+    dbRef: DbRef,
+    dbioRunner: DBIOActionRunner,
+    metricsRegistry: MetricRegistry,
+    resolvedDesignerConfig: Config,
+    // TODO: class wrapping all configs instead of raw hocon Config
+    featureTogglesConfig: FeatureTogglesConfig,
+    environment: String,
+    statisticsPublicKey: String,
+    futureProcessRepository: FetchingProcessRepository[Future],
+    scenarioActivityRepository: ScenarioActivityRepository,
+    scenarioLabelsRepository: ScenarioLabelsRepository,
+    globalNotificationRepository: InMemoryTimeseriesRepository[Notification],
+    feStatisticsRepository: FEStatisticsRepository[Future],
+    componentService: ComponentService,
+    processService: ProcessService,
+    fetchScenarioActivityService: FetchScenarioActivityService,
+    actionRepository: ScenarioActionRepository,
+    deploymentRepository: DeploymentRepository,
+    processChangeListener: ProcessChangeListener,
+    actionService: ActionService,
+    countsReporter: Option[CountsReporter[Future]],
+    counter: ProcessCounter,
+    fingerprintService: FingerprintService,
+    scenarioStatusProvider: ScenarioStatusProvider,
+    scenarioStatusPresenter: ScenarioStatusPresenter,
+    dmDispatcher: DeploymentManagerDispatcher,
+    processingTypeServicesProvider: ProcessingTypeDataProvider[ProcessingTypeServices, CombinedProcessingTypeData],
+    reloadProcessingTypes: () => IO[Unit],
+    processAuthorizer: AuthorizeProcess,
+    authenticationResources: AuthenticationResources,
+    authManager: AuthManager,
+    customHttpServiceProviders: Map[String, CustomHttpServiceProvider],
+)(
+    implicit system: ActorSystem,
+    executionContextWithIORuntime: ExecutionContextWithIORuntime,
+) extends LazyLogging {
+
+  import net.ceedubs.ficus.Ficus._
+  import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+
+  def createRoute: Route = {
```

Comment:
For me, this method is hard to read. First what I see are the details of creating each service. The first thing that interests me is how the app route is created. 

So, I'd recommend to convert these `val`s into private methods.

Reply:
For me too, I did something with this, take a look